common/stats/status_server.h

```diff
@@ -35,15 +39,40 @@ class Stats;
 
 class StatusServer {
  public:
+  /*
+   * \brief Endpoint/function name to std::function mapping.
+   *
+   * Stores the mapping of an end-point/function to the the
+   * actual std::function to be executed when requested.
+   *
+   * The std::function takes as input the list of arguments to
+   * the function in the form of a vector of [key, value] pairs
+   * (arrays can be passed in as delimited values) and returns a
+   * std::string which is expected to be (json/text) formatted.
+   */
   using EndPointToOPMap =
-      std::map<std::string, std::function<std::string(void)>>;
-
-  // Instantiate a StatusServer.
+      std::map<std::string, std::function<std::string(
+              const std::vector<std::pair<std::string, std::string>>* )>>;
+
+  /*!
+   * \brief Instantiate a StatusServer.
+   *
+   * The end-point handler in the EndPointToOPMap is an
+   * std::function that take as input the list of arguments to
+   * the function in the form of a vector of [key, value] pairs
+   * (arrays can be passed in as delimited values) and returns a
+   * std::string which is expected to be (json/text) formatted.
```

Comment:
Maybe remove the comment of "which is expected to be (json/text) formatted". It does not have much meaning now as the status server does not care about the format of the response.